### PR TITLE
feat: icons integration

### PR DESCRIPTION
**Pull request by gabyzif**
This PR is an integration branch used for testing before merging it to main. 

The main topic is Icons.

It handles the alignment of the text and icon inside the button:
[PR 1](https://github.com/mParticle/aquarium/pull/244 )
[PR 2](https://github.com/mParticle/aquarium/pull/243 )

The possibility to have an icon with a violet background and adds some missing svgs
[PR 3](https://github.com/mParticle/aquarium/pull/241
)

And it adds icon sizes
[PR 4](https://github.com/mParticle/aquarium/pull/238)

As a result, in this realese users will be able to

- Align icon and button using the variant `with-icon` inside the button

```ts
args: {
    icon: <Icon name="siteMap" size="xl" />,
    children: ' ',
    type: 'default',
    shape: 'round',
    variant: 'with-icon',
  },
```

add a background color to the icon using the property `type="background-solid"`

Screenshot 2024-05-24 at 7 58 28 PM

```ts
const mpLogoWithBackground: IGlobalNavigationLogo = {
  label: 'Data Platform',
  icon: 'catalog',
  type: 'background-solid',
  onSuiteLogoClick: () => {
    alert('Going to mP Home!')
  },
}
```

and choose different sizes of the icons with some new sizes like

`'xxxxl', 'xxxl', 'xxl', 'xl', 'lg', 'md', 'sm', 'xs'`

### CODE REVIEW

File: src/components/navigation/GlobalNavigation/GlobalNavigation.stories.tsx [hunk 1]
```diff
@@ -1063,3 +1083,69 @@ export const UseNewExperienceReminderHook: Story = {
...
+    },
+    tools: mpTools,
+    management: mpManagement,
+    orgs: mpOrgs,
+    minimapOptions: { overviewHref: '/', show: true },
+    onMpHomeClick: () => {
+      alert('going to overview map')
+    },
+    avatarOptions: {
+      // src: "https://static-qa1.qa.corp.mparticle.com/appimg/logo_af_916397d2-9732-8de6-77cc-80e3bba120ca.png",
```
[the remark this answers is not in the document]
Comment (gabyzif) [reply src/components/navigation/GlobalNavigation/GlobalNavigation.stories.tsx:1104]: I don't think so?

File: src/components/general/Button/Button.tsx [hunk 2]
```diff
@@ -2,12 +2,19 @@ import { Button as AntButton } from 'antd'
 import { type ButtonProps as AntButtonProps } from 'antd'
 import { ConfigProvider } from 'src/components/other/ConfigProvider/ConfigProvider'
 
-export interface IButtonProps extends AntButtonProps {}
-
+export interface IButtonProps extends AntButtonProps {
+  variant?: 'with-icon'
```
[the remark this answers is not in the document]
Comment (gabyzif) [reply src/components/general/Button/Button.tsx]: done :)

File: src/components/navigation/GlobalNavigation/GlobalNavigation.stories.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (tibuurcio) [reply src/components/navigation/GlobalNavigation/GlobalNavigation.stories.tsx:1104]: Old code, probably me 😅

**Review by tibuurcio — APPROVED** [review]
✨

**Pull request merged**